tests/lava/lib/optimization/utils/generators/test_mis.py

```diff
@@ -30,83 +30,50 @@ def test_num_vertices_prop(self):
-        self.assertEqual(graph.number_of_edges(), number_of_edges)
+        self.assertEqual(graph.number_of_nodes(), self.num_vertices)
+        self.assertEqual(2 * graph.number_of_edges(), self.num_edges)
 
     def test_get_complement_graph_matrix(self):
         """Tests the correct complement graph adjacency matrix is returned."""
         matrix = self.problem.get_complement_graph_matrix()
-        correct_matrix = np.array([[0, 1, 0, 0, 0, 0, 0, 1, 0, 0, ],
-                                   [1, 0, 1, 0, 0, 0, 0, 0, 0, 0, ],
-                                   [0, 1, 0, 0, 0, 1, 0, 0, 0, 0, ],
-                                   [0, 0, 0, 0, 1, 1, 0, 0, 0, 0, ],
-                                   [0, 0, 0, 1, 0, 0, 0, 0, 0, 0, ],
-                                   [0, 0, 1, 1, 0, 0, 0, 0, 0, 0, ],
-                                   [0, 0, 0, 0, 0, 0, 0, 1, 0, 1, ],
-                                   [1, 0, 0, 0, 0, 0, 1, 0, 0, 0, ],
-                                   [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ],
-                                   [0, 0, 0, 0, 0, 0, 1, 0, 0, 0, ]])
-        self.assertTrue((correct_matrix == matrix).all())
+        correct_matrix = np.array([[0, 0, 0, 1],
+                                   [0, 0, 1, 1],
+                                   [0, 1, 0, 0],
+                                   [1, 1, 0, 0]])
+        self.assertTrue((matrix == correct_matrix).all())
 
     def test_get_as_qubo(self):
         """Tests the conversion to QUBO returns the correct cost matrix."""
-        w_diag = 1.0
-        w_off = 4.0
+        w_diag = 1
+        w_off = 4
         qubo = self.problem.get_as_qubo(w_diag, w_off)
-        correct_matrix = np.array([[-1, 0, 2, 2, 2, 2, 2, 0, 2, 2],
-                                   [0, -1, 0, 2, 2, 2, 2, 2, 2, 2],
-                                   [2, 0, -1, 2, 2, 0, 2, 2, 2, 2],
-                                   [2, 2, 2, -1, 0, 0, 2, 2, 2, 2],
-                                   [2, 2, 2, 0, -1, 2, 2, 2, 2, 2],
-                                   [2, 2, 0, 0, 2, -1, 2, 2, 2, 2],
-                                   [2, 2, 2, 2, 2, 2, -1, 0, 2, 0],
-                                   [0, 2, 2, 2, 2, 2, 0, -1, 2, 2],
-                                   [2, 2, 2, 2, 2, 2, 2, 2, -1, 2],
-                                   [2, 2, 2, 2, 2, 2, 0, 2, 2, -1]])
+        correct_matrix = np.array([[-1, 2, 2, 0],
+                                   [2, -1, 0, 0],
+                                   [2, 0, -1, 2],
+                                   [0, 0, 2, -1]])
         self.assertTrue((correct_matrix == qubo.q).all())
 
     def test_find_maximum_independent_set(self):
         """Tests the correct maximum independent set is returned."""
         mis = self.problem.find_maximum_independent_set()
-        correct_mis = np.array([0, 0, 0, 0, 0, 0, 1, 0, 0, 1])
+        correct_mis = np.array([0, 1, 1, 0])
         self.assertTrue((correct_mis == mis).all())
 

```

Comment:
should the test below be on this same file? It seems those tests are actually testing the OptimizationSolver class.